development/build/scripts.js

```diff
@@ -471,6 +471,33 @@ function createScriptTasks({
         version,
         applyLavaMoat,
       }),
+      () => {
+        // MV3 injects inpage into the tab's main world, but in MV2 we need
+        // to do it manually:
+        if (process.env.ENABLE_MV3) {
+          return;
+        }
+        // stringify inpage.js into itself, and then make it inject itself into the page
+        browserPlatforms.forEach((browser) => {
+          const inpagePath = path.join(
+            __dirname,
+            '../../',
+            `dist/${browser}/${inpage}.js`,
+          );
+          const textContent = JSON.stringify(readFileSync(inpagePath, 'utf8'))
+            .replace(/\u2028/gu, '\\u2028')
```

Comment:
breakout wrapping and escaping into a utility function

.replace calls need individual comments explaining what they are doing

should replace be replaceAll ?

Reply:
The unicode escapes are no longer needed since Chrome 72 and Firefox 64 (p.s., the `g` flag in the replace regex makes it match _and replace_ all).

Per MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#well-formed_json.stringify

I've moved this out to its own utility function in https://github.com/MetaMask/metamask-extension/pull/22524/commits/89e9317a96e8d463a0e528c28214ddf8a73c5645